packages/cpp/src/browser/cpp-build-manager.ts

```diff
@@ -0,0 +1,62 @@
+/********************************************************************************
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { TaskService } from '@theia/task/lib/browser';
+import { CppBuildConfiguration, CppBuildTaskConfiguration } from './cpp-build-configurations';
+
+const parseArgv = require('string-argv');
+
+@injectable()
+export class CppBuildManager {
+
+    @inject(TaskService) protected readonly taskService: TaskService;
+
+    async runConfiguration(cppConfig: CppBuildConfiguration, options?: Partial<CppBuildTaskConfiguration>): Promise<void> {
+        const taskConfig = await this.getTaskConfiguration(cppConfig, options);
+        await this.taskService.runTask(taskConfig);
+    }
+
+    async getTaskConfiguration(cppConfig: CppBuildConfiguration, options: Partial<CppBuildTaskConfiguration> = {}): Promise<CppBuildTaskConfiguration> {
+        const commandString = cppConfig.command || await this.guessBuildCommand(cppConfig);
+        const argv: string[] = parseArgv(commandString || '');
+        if (argv.length === 0) {
+            throw new Error(`Invalid command in the build configuration "${cppConfig.name}"`);
+        }
+        const command: string = argv[0];
+        const args: string[] = argv.slice(1);
+        return {
+            type: 'cpp',
+            configuration: cppConfig,
+            cwd: cppConfig.directory,
+            label: cppConfig.name,
```

Comment:
I think the task name should contain build.  Build config names would usually look like `Release`, `Release`, `i386`, `Windows`, etc...  So maybe `cpp: Build ${config.name} configuration`?  It would give:

* `cpp: Build Release configuration`
* `cpp: Build Debug configuration`
* `cpp: Build i386 configuration`
* `cpp: Build Windows configuration`